### ISSUE TITLE
[Bug] Bug in get_simcc_maximum() of utils/post_processing.py

### ISSUE THREAD

**Issue by rtgunti** [open-mmlab/mmpose#2525]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmpose/issues) and [Discussions](https://github.com/open-mmlab/mmpose/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmpose).

### Environment

python -c "from mmpose.utils import collect_env; print(collect_env())"

OrderedDict([('sys.platform', 'linux'), ('Python', '3.8.16 (default, Mar  2 2023, 03:21:46) [GCC 11.2.0]'), ('CUDA available', True), ('numpy_random_seed', 2147483648), ('GPU 0', 'Tesla K80'), ('CUDA_HOME', None), ('GCC', 'gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0'), ('PyTorch', '1.11.0'), ('PyTorch compiling details', 'PyTorch built with:\n  - GCC 7.3\n  - C++ Version: 201402\n  - Intel(R) oneAPI Math Kernel Library Version 2023.1-Product Build 20230303 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.5.2 (Git Hash a9302535553c73243c632ad3c4c80beec3d19a1e)\n  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - LAPACK is enabled (usually provided by MKL)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 11.3\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37\n  - CuDNN 8.2\n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.11.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, \n'), ('TorchVision', '0.12.0'), ('OpenCV', '4.8.0'), ('MMEngine', '0.8.1'), ('MMPose', '1.1.0+537bd8e')])
---------  ---------  --------------------------------------
mim list
Package    Version    Source
---------  ---------  --------------------------------------
mmcv       2.0.0rc3   https://github.com/open-mmlab/mmcv
mmcv-full  1.7.1      https://github.com/open-mmlab/mmcv
mmengine   0.4.0      https://github.com/open-mmlab/mmengine

### Reproduces the problem - code sample

```python   
mask = max_val_x > max_val_y
max_val_x[mask] = max_val_y[mask]
vals = max_val_x
```

### Reproduces the problem - command or script

```shell
python tools/train configs/body_2d_keypoint/rtmpose/coco/rtmpose-l_8xb256-420e_aic-coco-256x192.py
```

### Reproduces the problem - error message

Not Applicable

### Additional information

File: mmpose/codecs/utils/post_processing.py
Method: [get_simcc_maximum()](https://github.com/open-mmlab/mmpose/blob/537bd8e543ab463fb55120d5caaa1ae22d6aaf06/mmpose/codecs/utils/post_processing.py#L85C4-L85C4)

The goal is to get maximum values from both max_val_x and max_val_y for each corresponding element. However, the mask generated seems to be incorrect. 

Observed : mask = max_val_x > max_val_y
Proposed : mask = max_val_x < max_val_y

This way, the lower vals in max_val_x are replaced with higher vals of max_val_y.

**Comment by Tau-J** [comment]
Hi @rtgunti , thanks for using MMPose. This code is from the [official repo](https://github.com/leeyegy/SimCC/blob/aa6e089af29718e889680b6226eb7cac107aa21a/lib/core/function.py#L155). Personally, I think you are right, yet the performance on COCO dropped about 0.3 AP after changing the post processing. Therefore, maybe we can just keep it. I think use the maximum value of the predicted distribution as scores may be too confident, which results in negative impacts on nms of evaluation.
